lib/remote.go

```diff
@@ -52,25 +47,17 @@ func (m RemoteMethods) Feeds(ctx context.Context, p *FeedsParams) (map[string][]
 
 // PreviewParams provides arguments to the preview method
 type PreviewParams struct {
-	Remote string
-	Ref    string
+	Ref string
 }
 
 // UnmarshalFromRequest implements a custom deserialization-from-HTTP request
 func (p *PreviewParams) UnmarshalFromRequest(r *http.Request) error {
 	if p == nil {
 		p = &PreviewParams{}
 	}
-
-	params := *p
-	if params.Ref == "" {
-		params.Ref = r.FormValue("refstr")
-	}
-	if params.Remote == "" {
-		params.Remote = r.FormValue("remote")
+	if p.Ref == "" {
+		p.Ref = r.FormValue("refstr")
```

Comment:
Should we also change `r.FormValue("refstr")` > `r.FormValue("ref")` ~~and have the middleware reflect that too | I now checked and that part is done so we should just make sure all `refstr` are converted to `ref`~~?

Reply:
Ah, I missed this one, good call!